akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -141,25 +120,42 @@ abstract class EventSourcedBehavior[Command, Event, State >: Null] private[akka]
    */
   def snapshotSelectionCriteria: SnapshotSelectionCriteria = SnapshotSelectionCriteria.latest
 
+  /**
+   * Initiates a snapshot if the given function returns true.
+   * When persisting multiple events at once the snapshot is triggered after all the events have
+   * been persisted.
+   *
+   * receives the State, Event and the sequenceNr used for the Event
+   *
+   * @return `true` if snapshot should be saved for the given event
+   * @see [[EventSourcedBehavior#snapshotEvery]]
+   */
+  def shouldSnapshot(@unused state: State, @unused event: Event, @unused sequenceNr: Long): Boolean = false
```

Comment:
Wait, how does this relate to also defining a `retentionCriteria`? Shouldn't it be only that and skip this?

Reply:
By default false?